redbot/cogs/alias/alias.py

```diff
@@ -287,22 +288,48 @@ async def _del_global_alias(self, ctx: commands.Context, alias_name: str):
 
     @alias.command(name="list")
     @commands.guild_only()
+    @checks.bot_has_permissions(add_reactions=True)
     async def _list_alias(self, ctx: commands.Context):
         """List the available aliases on this server."""
         guild_aliases = await self._aliases.get_guild_aliases(ctx.guild)
         if not guild_aliases:
             return await ctx.send(_("There are no aliases on this server."))
-        names = [_("Aliases:")] + sorted(["+ " + a.name for a in guild_aliases])
-        await ctx.send(box("\n".join(names), "diff"))
+        names = [""] + sorted(["+ " + a.name for a in guild_aliases])
+        message = ""
+        for a in names:
+            message += "{}\n".format(a)
+        temp = list(pagify(message, delims=["\n"], page_length=1850))
+        alias_list = []
+        count = 0
+        for page in temp:
+            count += 1
+            page = (_("Aliases:")) + page + ("\n\nPage {}/{}".format(count, len(temp)))
+            alias_list.append(box("".join(page), "diff"))
+        if len(alias_list) == 1:
+            return await ctx.send(alias_list[0])
+        await menu(ctx, alias_list, DEFAULT_CONTROLS)
 
     @global_.command(name="list")
+    @checks.bot_has_permissions(add_reactions=True)
     async def _list_global_alias(self, ctx: commands.Context):
         """List the available global aliases on this bot."""
         global_aliases = await self._aliases.get_global_aliases()
         if not global_aliases:
             return await ctx.send(_("There are no global aliases."))
-        names = [_("Aliases:")] + sorted(["+ " + a.name for a in global_aliases])
-        await ctx.send(box("\n".join(names), "diff"))
+        names = [""] + sorted(["+ " + a.name for a in global_aliases])
+        message = ""
+        for a in names:
+            message += "{}\n".format(a)
```

Comment:
This can be simplified by using `.join()` as it were before the change. This will also get rid of the trailing `\n` that makes the last page look different.
```suggestion
        message = "\n".join(names)
```